tests/drizzle/test_kernel.py

```diff
@@ -0,0 +1,140 @@
+import pytest
+import numpy as np
+from astropy import wcs
+from drizzlepac import cdriz
+import cdriz_setup
+
+
+@pytest.fixture
+def kernel_pars():
+    _params = cdriz_setup.Get_Grid(inx=50, iny=60, outx=51, outy=66)
+    _params.zero_background()
+    return _params
+
+
+@pytest.mark.parametrize("kernel", ["square", "point", "turbo", "gaussian", "lanczos3"])
+def test_point_kernel(kernel, kernel_pars, new_truth=False, return_png=False):
+    """Function tests different c code point kernels (inputs already created on instantiation).
+
+    Parameters
+    ----------
+    kernel : str
+        String associated with one of the c code point kernel options.
+    kernel_pars : Class
+        The Class inintialized in Get_Grid which includes all of the inputs need to run cdriz.tdriz.
+    new_truth : bool, optional
+        Flag, whether to save a new csv truth file from the output data, by default False
+    return_png: bool, optional
+        Flag, whether to create an output image map (png).
+    """
+    # truth filename
+    truth_filename = f"./tests/drizzle/truth_files/{kernel}_truth"
+
+    # add missing/flagged pixels in inwht
+    kernel_pars.insci[20:22, 21:22] = 100
+
+    # resample:
+    cdriz_setup.cdriz_call(kernel_pars, kernel)
+
+    if new_truth:
+        # save new truth file
+        cdriz_setup.save_array(kernel_pars.outsci, f"{truth_filename}.csv")
```

Comment:
The code doesn't actually use this new_truth flag with that name. The names for new truth files (if they differ from the old ones) is specified in the assert statements with a "_new" at the end.  

I've removed the above if statement and added try/except elements so that the code creates new truth files with the regular name (not with "_new") when the truth files are missing. This should make create new truth files easier.